Asgard/Communications/EventArgs/CbusMessageEventArgs.cs

```diff
@@ -5,13 +5,14 @@ namespace Asgard.Communications
 {
     public class CbusMessageEventArgs : EventArgs
     {
-        public ICbusMessage Message { get; }
-
+        public ICbusMessage? Message { get; }
+        public string? GridConnectMessage { get; }
         public bool Received { get; }
 
-        public CbusMessageEventArgs(ICbusMessage message, bool received)
+        public CbusMessageEventArgs(ICbusMessage? message, string? gridConnectMessage, bool received)
         {
             this.Message = message;
+            this.GridConnectMessage = gridConnectMessage;
```

Comment:
I'm not a fan of this addition - this layer serves to abstract away the underlying gridconnect protocol.

Reply:
If I remember correctly I put that in so it was possible to log the received message in one place regardless of whether it could be resolved to a CBUS OpCode or not. If it wasn't a CBUS OpCode it logged the GridConnectMessage instead. It might well be better to handle this via ToString(). I'll take a look over the weekend.